rdrf/registry/patients/admin_forms.py

```diff
@@ -212,16 +212,75 @@ def clean_signature(self):
 
 
 class PatientStageForm(forms.ModelForm):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        if 'instance' in kwargs and kwargs['instance'] is not None:
+            instance = kwargs['instance']
+            stages_qs = PatientStage.objects.filter(registry=instance.registry)
+            self.fields["allowed_prev_stages"].queryset = stages_qs
+            self.fields["allowed_next_stages"].queryset = stages_qs
+            self.fields["registry"].disabled = True
+
+    def clean(self):
+        cleaneddata = self.cleaned_data
```

Comment:
I think the usual pattern is to use:

`cleaneddata = super().clean()`

and then you can modify `cleaneddata` as required and return that at the end of the method.